### PR TITLE
[parity-crypto] prepare 0.5.0

### PR DESCRIPTION
**Pull request by dvdplm**

### CODE REVIEW

File: parity-crypto/CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,11 @@ The format is based on [Keep a Changelog].
...
 
 ## [Unreleased]
-- Remove `inv()` from `SecretKey` (breaking)
-- `Generate::generate()` does not return error
-- `Secp256k1` is no longer exported 
-- Remove `public_is_valid()` as it is now impossible to create invalid public keys
-- 0-valued `Secp::Message`s are disallowed (signatures on them are forgeable for all keys)
-- updates to upstream `rust-secp256k1` at v0.17.2
+
+## [0.5.0] - 2020-01-08
```
Comment (ordian) [inline parity-crypto/CHANGELOG.md]:
```suggestion
## [0.5.0] - 2020-02-08
```

**Comment by niklasad1** [comment]
Can we include #337 in this release?

**Comment by dvdplm** [comment]
> Can we include #337 in this release?

I jumped the gun on that, apologies. `rustc-hex` is actually used in `parity-crypto` when the `publickey` feature is active.

**Comment by ordian** [comment]
> > Can we include #337 in this release?
> 
> I jumped the gun on that, apologies. `rustc-hex` is actually used in `parity-crypto` when the `publickey` feature is active.

Ah, good catch. Indeed https://github.com/paritytech/parity-common/blob/2c5273c2e983aa6398cd11d3af65c3d3cf7e6851/parity-crypto/src/publickey/ecdsa_signature.rs#L21 
It should be optional then and included in the `publickey` feature set.

I assumed our tests would have caught that, would be nice to add `cargo check -p parity-crypto --all-features`

**Comment by niklasad1** [comment]
> I jumped the gun on that, apologies. rustc-hex is actually used in parity-crypto when the publickey feature is active.

sorry, missed that. let's revert #337

**Pull request merged**